app/services/offer_validations.rb

```diff
@@ -41,4 +44,28 @@ def ratifying_provider_changed?
       errors.add(:base, :different_ratifying_provider)
     end
   end
+
+  def restrict_reverting_rejection
+    if application_choice.rejected_by_default
+      errors.add(:base, :application_rejected_by_default)
+    end
+
+    if !candidate_in_apply_2? && application_choice.application_form.apply_1? && any_accepted_offers?
+      errors.add(:base, :other_offer_already_accepted)
+    end
+
+    if candidate_in_apply_2? && !application_choice.candidate.current_application.eql?(application_choice)
+      errors.add(:base, :only_latest_application_rejection_can_be_reverted_on_apply_2)
+    end
+  end
+
+private
+
+  def any_accepted_offers?
+    ((application_choice.self_and_siblings - [application_choice]).map(&:status).map(&:to_sym) & ApplicationStateChange::ACCEPTED_STATES).any?
```

Comment:
This is a method on application choice; there is no `siblings` method. It could potentially be refactored and moved there but since we only use it once I wouldn't rush into it. The actual implementation is

```ruby
def self_and_siblings
   application_form.application_choices
end
```

Reply:
ah I thought we were using the rails methods, nvm then